x-pack/plugins/security/server/lib/authentication/providers/basic.ts

```diff
@@ -15,13 +18,13 @@ export class BasicCredentials {
   /**
    * Takes provided `username` and `password`, transforms them into proper `Basic ***` authorization
    * header and decorates passed request with it.
-   * @param {Hapi.Request} request HapiJS request instance.
-   * @param {string} username User name.
-   * @param {string} password User password.
-   * @returns {Hapi.Request} HapiJS request instance decorated with the proper header.
+   * @param request HapiJS request instance.
+   * @param username User name.
+   * @param password User password.
    */
-  static decorateRequest(request, username, password) {
-    if (!request || typeof request !== 'object') {
+  public static decorateRequest<T extends Request>(request: T, username: string, password: string) {
+    const typeOfRequest = typeof request;
```

Comment:
__note:__ that's just to trick Typescript type-guard system, found neither more elegant solution nor TS issue about that :/ In short if it sees `typeof request` in `if` below then if we don't throw `request` will have type `never`.